src/tests/Microsoft.PowerFx.Core.Tests/CheckResultTests.cs

```diff
@@ -280,5 +282,47 @@ public void TestIRFail()
 
             Assert.Throws<InvalidOperationException>(() => check.ApplyIR());
         }
+
+        // CheckResult properly wired up to invariant translator. 
+        // More tests at DisplayNameTests
+        [Fact]
+        public void TestApplyGetInvariant()
+        {
+            var check = new CheckResult(new Engine());
+
+            Assert.Throws<InvalidOperationException>(() => check.ApplyGetInvariant());
+
+            var r1 = RecordType.Empty()
+              .Add(new NamedFormulaType("new_field", FormulaType.Number, "Field"));
+
+            // display name: Field --> new_field
+            // lexer locale: 2,3 --> 2.3
+            check.SetText("Field + 2,3", _frCultureOpts);
+            Assert.Throws<InvalidOperationException>(() => check.ApplyGetInvariant());
+
+            check.SetBindingInfo(r1);
+            var invariant = check.ApplyGetInvariant();
+
+            Assert.Equal("new_field + 2.3", invariant);
+        }
+
+        // CheckResult properly wired up to Apply logging. 
+        [Theory]
+        [InlineData("123+abc", "#$number$# + #$firstname$#", true)] // display names
+        [InlineData("123+", "#$number$# + #$error$#", false)] // error 
+        [InlineData("123,456", "#$number$#", true)] // locales 
+        [InlineData("Power(2,3)", "Power(#$number$#)", true)] // functions aren't Pii
```

Comment:
>[InlineData("Power(2,3)", "Power(#$number$#)", true)] [](http://example.com/codeflow?start=8&length=53)

Not within the scope of this change but shouldn't the expected result be "Power(#$number$#,#$number$#)"?

Reply:
yeah, I'd expect that too - had to update the test.